pkg/channel/lookup.go

```diff
@@ -328,3 +334,20 @@ func NewMoreThanOneTemplateCandidateErr(component v1beta2.Module,
 	return fmt.Errorf("%w: more than one module template found for module: %s, candidates: %v",
 		ErrTemplateNotIdentified, component.Name, candidates)
 }
+
+func saveDescriptorToCache(template ModuleTemplateTO) error {
+	if template.ModuleTemplate == nil {
+		return ErrModuleTemplateIsNil
+	}
+	descriptor := template.GetDescFromCache()
+	if descriptor == nil {
```

Comment:
still the coding style, it's better to always deal with unexpected behavior first, so that you can avoid nested if condition

```
if descriptor != nil {
  return nil
}

//the rest code 

```